### PR TITLE
DOC: stats: example treatment odd_ratio

### PR DESCRIPTION
**Pull request by tupui**
Replace a made up example using the same example as in #17610.

This was suggested in https://github.com/scipy/scipy/pull/17610/files#r1051034735 as a good complementary analysis for this case.

### CODE REVIEW

File: scipy/stats/_odds_ratio.py [hunk 1]
```diff
@@ -376,6 +376,10 @@ def odds_ratio(table, *, kind='conditional'):
     .. [3] H. Sahai and A. Khurshid (1996), Statistics in Epidemiology:
            Methods, Techniques, and Applications, CRC Press LLC, Boca
            Raton, Florida.
+    .. [4] Berger, Jeffrey S. et al. "Aspirin for the Primary Prevention of
+           Cardiovascular Events in Women and Men: A Sex-Specific
+           Meta-analysis of Randomized Controlled Trials."
+           JAMA, 295(3):306–313, :doi:`10.1001/jama.295.3.306`, 2006.
```
Comment (tupui) [inline scipy/stats/_odds_ratio.py:382]:
```suggestion
           JAMA, 295(3):306-313, :doi:`10.1001/jama.295.3.306`, 2006.
```

File: scipy/stats/_odds_ratio.py [hunk 2]
```diff
@@ -402,36 +406,45 @@ def odds_ratio(table, *, kind='conditional'):
...
 
     We can compute the 95% confidence interval for the odds ratio:
 
     >>> res.confidence_interval(confidence_level=0.95)
-    ConfidenceInterval(low=1.2514829132266785, high=10.363493716701269)
+    ConfidenceInterval(low=0.6241234078749812, high=0.9354102892100372)
 
     The 95% confidence interval for the conditional odds ratio is
-    approximately (1.25, 10.4).
+    approximately (0.62, 0.94).
```
Comment (mdhaber) [inline scipy/stats/_odds_ratio.py:447]:
```suggestion
    approximately (0.62, 0.94).

The fact that the entire 95% confidence interval falls below 1 supports
the authors' conclusion that the aspirin was associated with a statistically
significant reduction in stroke at the 5% significance level.
```
Although this sort of statement is usually true for, say, confidence intervals based on normal approximation, I'm not sure if this is strictly true for the exact confidence intervals calculated for the odds ratio. The reason is that the shape of the test statistic null distribution changes as the true odds ratio changes. That is, the width of the 95% confidence interval is not the same when the odds ratio is not 1 as when the null hypothesis odds ratio is 1. Perhaps we should change this to something less specific - just remove the part about 5% significance level.

Comment (mdhaber) [reply scipy/stats/_odds_ratio.py:447]: I'm removing the part about 5% significance level.

Comment (mdhaber) [reply scipy/stats/_odds_ratio.py:447]:
```suggestion
    approximately (0.62, 0.94).

The fact that the entire 95% confidence interval falls below 1 supports
the authors' conclusion that the aspirin was associated with a statistically
significant reduction in ischemic stroke.```

File: scipy/stats/_odds_ratio.py [hunk 3]
```diff
@@ -402,36 +406,50 @@ def odds_ratio(table, *, kind='conditional'):
...
+
+The fact that the entire 95% confidence interval falls below 1 supports
+the authors' conclusion that the aspirin was associated with a statistically
+significant reduction in ischemic stroke.```
     """
```
Comment (tupui) [inline scipy/stats/_odds_ratio.py:453]:
```suggestion

    The fact that the entire 95% confidence interval falls below 1 supports
    the authors' conclusion that the aspirin was associated with a
    statistically significant reduction in ischemic stroke.
    """
```

**Comment by tupui** [comment]
Thank you for working on this Matt. I just resolved the CI issue (let's see if we are green).

File: scipy/stats/_odds_ratio.py [hunk 4]
```diff
@@ -402,36 +406,50 @@ def odds_ratio(table, *, kind='conditional'):
...
-    exposure to a certain chemical is assocated with increased occurrence
-    of a certain disease.  Suppose we have the following table for a
-    collection of 410 people::
+    In [4]_, the use of aspirin to prevent cardiovascular events in women
+    and men was investigated. The study notably concluded:
 
-                  exposed   unexposed
-        cases         7         15
-        noncases     58        472
+        For women and men, aspirin therapy reduced the risk of a composite of
```
Comment (mdhaber) [inline scipy/stats/_odds_ratio.py:412]:
```suggestion
        ...aspirin therapy reduced the risk of a composite of
```

File: scipy/stats/_odds_ratio.py [hunk 5]
```diff
@@ -402,36 +406,50 @@ def odds_ratio(table, *, kind='conditional'):
...
+        cardiovascular events due to its effect on reducing the risk of
+        ischemic stroke in women [...]
 
-    The question we ask is "Is exposure to the chemical associated with
-    increased risk of the disease?"
+    The article list multiple studies for various cardiovascular events. Let's
```
Comment (mdhaber) [inline scipy/stats/_odds_ratio.py:416]:
```suggestion
    The article lists studies of various cardiovascular events. Let's
```

File: scipy/stats/_odds_ratio.py [hunk 6]
```diff
@@ -402,36 +406,50 @@ def odds_ratio(table, *, kind='conditional'):
...
+    focus on the ischemic stoke in women.
+
+    The following table summarizes the results of the experiment in which
+    participants took aspirin or a placebo on a regular basis for several
+    years. Occurences of ischemic stroke were recorded::
```
Comment (mdhaber) [inline scipy/stats/_odds_ratio.py:421]:
```suggestion
    years. Cases of ischemic stroke were recorded::
```

File: scipy/stats/_odds_ratio.py [hunk 7]
```diff
@@ -402,36 +406,50 @@ def odds_ratio(table, *, kind='conditional'):
...
+    >>> res = odds_ratio([[176, 230], [21035, 21018]])
     >>> res.statistic
-    3.7836687705553493
+    0.7646037659999126
 
-    For this sample, the odds of getting the disease for those who have
-    been exposed to the chemical are almost 3.8 times that of those who
-    have not been exposed.
+    For this sample, the odds of getting an ischemic stroke for those who have
+    been taking aspiring to the chemical are 0.76 times that of those
```
Comment (mdhaber) [inline scipy/stats/_odds_ratio.py:438]:
```suggestion
    been taking aspirin are 0.76 times that of those
```

**Review by mdhaber — APPROVED** [review]

**Pull request merged**